server/src/main/java/org/opensearch/action/admin/cluster/snapshots/restore/RestoreSnapshotRequest.java

```diff
@@ -146,7 +145,7 @@ public RestoreSnapshotRequest(StreamInput in) throws IOException {
         indexSettings = readSettingsFromStream(in);
         ignoreIndexSettings = in.readStringArray();
         snapshotUuid = in.readOptionalString();
-        if (FeatureFlags.isEnabled(FeatureFlags.SEARCHABLE_SNAPSHOT) && in.getVersion().onOrAfter(Version.V_2_4_0)) {
+        if (in.getVersion().onOrAfter(Version.V_2_4_0)) {
```

Comment:
Due to the feature flag, 2.4 doesn't actually understand this field (by default). I think we'll have to change this to 3.0 in this PR, then change it to 2.7 after the backport.

Reply:
It does mean the released version will be incompatible with the pre-release feature-flagged versions, which I think is probably fine.

Reply:
Yep. I think we also need to update the `V_3_0_0` references with `V_2_7_0` later to have the correct BWC.